src/lib/crudAppointment.js

```diff
@@ -0,0 +1,195 @@
+
+// Function to open the modal to be inserted in the + button (new appointment)
+export function openModal() {
+  const modal = document.getElementById("myModal");
+  modal.style.display = "block";
+}
+
+// Function to close the modal when clicking outside the modal content
+window.onclick = function (event) {
+  const modal = document.getElementById("myModal");
+  if (event.target == modal) {
+    modal.style.display = "none";
+  }
+};
+
+// Generate a unique ID
+function generateId() {
+  return Math.floor(Math.random() * 999999);
+}
+
+// User-generated data extracted from HTML page
+const DESCRIPTION_KEY = "description";
+const DUEDATE_KEY = "dueDate"; // appointment due date
+const CATEGORY_KEY = "category";
+const USERID_KEY = "userId";
+
+// Retrieve data from HTML form
+export function getDataFromForm() {
+  const description = document.getElementById(DESCRIPTION_KEY).value;
+  const dueDate = new Date(document.getElementById(DUEDATE_KEY).value);
+  const category = document.getElementById(CATEGORY_KEY).value;
+
+  // Retrieve user ID from localStorage
+  const userId = localStorage.getItem(USERID_KEY);
+
+  // Generate appointment ID
+  const id = generateId();
+
+  // Create creation date
+  const creationDate = new Date();
+
+  // Initially, updateDate and completionDate are null
+  const updateDate = null;
+  const completionDate = null;
+
+  // Create appointment
+  createAppointment(
+    id,
+    userId,
+    description,
+    creationDate,
+    updateDate,
+    completionDate,
+    dueDate,
+    category
+  );
+}
+
+// Event listener to call getDataFromForm when the form is submitted
+document
+  .getElementById("yourFormId") // INSERT CORRECT HTML ID !!!!!!!!!!!!!!!!!!
+  .addEventListener("submit", function (event) {
+    event.preventDefault();
+    getDataFromForm();
+  });
+
+// Create appointment
+export function createAppointment(
+  id,
+  userId,
+  description,
+  creationDate,
+  updateDate,
+  completionDate,
+  dueDate,
+  category
+) {
+  const appointment = new Appointment(
+    id,
+    userId,
+    description,
+    creationDate,
+    updateDate,
+    completionDate,
+    dueDate,
+    category
+  );
+  localStorage.setItem(appointment.id, JSON.stringify(appointment));
+  displayAppointments(); // Update user interface after creating an appointment
+}
+
+// Update appointment
+export function updateAppointment(id, newValues) {
+  const appointment = JSON.parse(localStorage.getItem(id));
+  if (appointment) {
+    for (let key in newValues) {
+      if (appointment.hasOwnProperty(key)) {
+        appointment[key] = newValues[key];
+      }
+    }
+    appointment.updateDate = new Date();
+    localStorage.setItem(appointment.id, JSON.stringify(appointment));
+    displayAppointments(); // Update user interface after modifying an appointment
+  } else {
+    displayMessage("Appointment not found"); // Show a message to the user if the appointment is not found
+  }
+}
```

Comment:
1) this function should be responsible only of updating the value inside the localstorate, not update the ui
2) this function should have only an object of type Appointment as paramenter
3) the for loop should be remove, use a map instead like this:
const appointments = getAppointmentsFromLocalStorage()
saveAppointments(appointments.map(appointment => appointment.id === newAppointment.id ? newAppointment : appointment))